pytorch_lightning/trainer/properties.py

```diff
@@ -185,7 +185,21 @@ def progress_bar_dict(self) -> dict:
         """ Read-only for progress bar metrics. """
         ref_model = self.model if not self.data_parallel else self.model.module
         ref_model = cast(LightningModule, ref_model)
-        return dict(**ref_model.get_progress_bar_dict(), **self.logger_connector.progress_bar_metrics)
+
+        standard_metrics = ref_model.get_progress_bar_dict()
+        logged_metrics = self.logger_connector.progress_bar_metrics
+        duplicates = list(standard_metrics.keys() & logged_metrics.keys())
+        if duplicates:
+            rank_zero_warn(
+                f"The progress bar already tracks a metric with the name '{duplicates[0]}' and"
+                f" `self.log('{duplicates[0]}', ..., prog_bar=True)` will overwrite this value. "
+                f" If this is undesired, change the name or override `get_progress_bar_dict()`."
+                f" in LightingModule.",
+                UserWarning
```

Comment:
alternatively, one could drop this warning and just overwrite the default metric with what the user logged, since it is deliberate.

Reply:
@rohitgr7 if the user wants their logged value in the progress bar, there would be no way to turn off the warning. That's the main argument against the warning, because the user is deliberately choosing to display that value with `prog_bar=True`.